### PR TITLE
Merge APIM Gateway `3.5.x`

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

https://github.com/gravitee-io/issues/issues/6140

**Description**

Merge APIM Gateway 3.5.x inside the APIM monorepo

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Review by ThibaudAV — APPROVED** [review]
🤞

File: .circleci/config.yml [hunk 1]
```diff
@@ -208,17 +208,31 @@ jobs:
...
             - get-apim-tag
             -   run:
-                    name: Build & Publish Management API Docker Image to Azure Registry
+                    name: Build & Publish Management API and Gateway Docker Image to Azure Registry
                     command: |
                         export APIM_VERSION=$(mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)
-                        export IMAGE_TAG=graviteeio.azurecr.io/apim-management-api:${TAG}
+
+                        export REST_API_IMAGE_TAG=graviteeio.azurecr.io/apim-management-api:${TAG}
+                        export GATEWAY_IMAGE_TAG=graviteeio.azurecr.io/apim-management-api:${TAG}
```
Comment (phiz71) [inline .circleci/config.yml]:
```suggestion
                        export GATEWAY_IMAGE_TAG=graviteeio.azurecr.io/apim-gateway:${TAG}
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -208,17 +208,31 @@ jobs:
...
-                        -t ${IMAGE_TAG} .
+                        -t ${REST_API_IMAGE_TAG} .
+
+                        docker build -f gravitee-apim-gateway/docker/Dockerfile-dev \
+                        --build-arg GRAVITEEIO_VERSION=${APIM_VERSION} \
+                        -t ${GATEWAY_IMAGE_TAG} .
+
+                        if [[ $CIRCLE_BRANCH =~ ^([0-9]+\.[0-9]+\.x|master)$ ]]
+                        then
+                          docker login --username="${DOCKERHUB_BOT_USER_NAME}" -p="${DOCKERHUB_BOT_USER_TOKEN}"
```
Comment (phiz71) [inline .circleci/config.yml]: To fix, like on master

File: .circleci/config.yml [hunk 3]
```diff
@@ -368,6 +382,13 @@ jobs:
 
                         export IMAGE_TAG=graviteeio.azurecr.io/apim-console-webui:${TAG}
                         docker build -f docker/Dockerfile-dev -t ${IMAGE_TAG} .
+
+                        if [[ $CIRCLE_BRANCH =~ ^([0-9]+\.[0-9]+\.x|master)$ ]]
```
Comment (phiz71) [inline .circleci/config.yml]: To fix, like on master

File: .circleci/config.yml [hunk 4]
```diff
@@ -497,6 +518,13 @@ jobs:
 
                         export IMAGE_TAG=graviteeio.azurecr.io/apim-portal-webui:${TAG}
                         docker build -f docker/Dockerfile-dev -t ${IMAGE_TAG} .
+
+                        if [[ $CIRCLE_BRANCH =~ ^([0-9]+\.[0-9]+\.x|master)$ ]]
```
Comment (phiz71) [inline .circleci/config.yml]: To fix, like on master

File: .circleci/config.yml [hunk 5]
```diff
@@ -540,7 +568,9 @@ jobs:
                                          --set "portal.image.repository=graviteeio.azurecr.io/apim-portal-webui" \
                                          --set "portal.image.tag=${TAG}" \
                                          --set "ui.image.repository=graviteeio.azurecr.io/apim-console-webui" \
-                                         --set "ui.image.tag=${TAG}"
+                                         --set "ui.image.tag=${TAG}" \
+                                         --set "gateway.image.repository=graviteeio.azurecr.io/apim-gateway" \
+                                         --set "gateway.image.tag=${TAG}"
```
Comment (phiz71) [inline .circleci/config.yml]: Don't the forget the "hack" with redis-repository
https://github.com/gravitee-io/gravitee-gateway/blob/3.13.x/.circleci/config.yml#L210

File: .circleci/config.yml [hunk 6]
```diff
@@ -560,7 +590,15 @@ jobs:
                                      --set "portal.image.repository=graviteeio.azurecr.io/apim-portal-webui" \
                                      --set "portal.image.tag=${TAG}" \
                                      --set "ui.image.repository=graviteeio.azurecr.io/apim-console-webui" \
-                                     --set "ui.image.tag=${TAG}"
+                                     --set "ui.image.tag=${TAG}" \
+                                     --set "gateway.image.repository=graviteeio.azurecr.io/apim-gateway" \
+                                     --set "gateway.image.tag=${TAG}"
```
Comment (phiz71) [inline .circleci/config.yml]: Don't forget the "hack" with redis-repository
https://github.com/gravitee-io/gravitee-gateway/blob/3.13.x/.circleci/config.yml#L210

File: gravitee-apim-gateway/GIT_COMMIT [hunk 7]
```diff
@@ -0,0 +1 @@
+565d72f88a868053132f2750f3faeb2a2fb142f6
```
Comment (phiz71) [inline gravitee-apim-gateway/GIT_COMMIT]: necessary ?

**Review by phiz71 — CHANGES_REQUESTED** [review]

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/merge-3-5-x/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**